benchmark_test.go

```diff
@@ -609,3 +619,217 @@ func Benchmark_S3_segupload(b *testing.B) {
+
+	for _, str := range strs {
+		bloom.Add(str)
+	}
+
+	fname := "bloomfilter.bin"
+	file, err := os.OpenFile(fname, os.O_WRONLY|os.O_CREATE, 0644)
+	if err != nil {
+		panic(err)
+	}
+	defer file.Close()
+
+	writer := bufio.NewWriter(file)
+
+	_, err = bloom.WriteTo(writer)
+	if err != nil {
+		panic(err)
+	}
+
+	err = writer.Flush()
+	if err != nil {
+		panic(err)
+	}
+}
+
+func benchmarkCuckooLinvon(strs [][]byte) {
+	// Configs: https://github.com/linvon/cuckoo-filter
+	tagsPerBucket := uint(4)      // b in paper (entries per bucket)
+	bitsPerItem := uint(8)        // f in paper (fingerprint)
+	maxNumKeys := uint(len(strs)) // num of keys the filter will store
+
+	cuckoo := cuckooLin.NewFilter(tagsPerBucket, bitsPerItem, maxNumKeys, cuckooLin.TableTypePacked)
+
+	for _, str := range strs {
+		cuckoo.Add(str)
+	}
+
+	// Open a file for writing
+	fname := "cuckoo_filter_lin.bin"
+	file, err := os.OpenFile(fname, os.O_WRONLY|os.O_CREATE, 0644)
+	if err != nil {
+		// Handle error
+		panic(err)
+	}
+	defer file.Close()
+
+	writer := bufio.NewWriter(file)
+
+	data, err := cuckoo.Encode()
+	if err != nil {
+		panic(err)
+	}
+
+	_, err = writer.Write(data)
+	if err != nil {
+		panic(err)
+	}
+
+	err = writer.Flush()
+	if err != nil {
+		panic(err)
+	}
+}
+
+func benchmarkCuckooSeiflotfy(strs [][]byte) {
+	// Default config: https://github.com/seiflotfy/cuckoofilter
+	// b (bucket size): 4
+	// f (fingerprint): 8
+	// False positive rate: 0.03
+	numKeys := uint(len(strs)) // num of keys the filter will store
+	cuckoo := cuckooSeif.NewFilter(numKeys)
+
+	for _, str := range strs {
+		cuckoo.Insert(str) // Also try insert unique
+	}
+
+	fname := "cuckoo_filter_seif.bin"
+	file, err := os.OpenFile(fname, os.O_WRONLY|os.O_CREATE, 0644)
+	if err != nil {
+		panic(err)
+	}
+	defer file.Close()
+
+	writer := bufio.NewWriter(file)
+
+	data := cuckoo.Encode()
+
+	_, err = writer.Write(data)
+	if err != nil {
+		panic(err)
+	}
+
+	err = writer.Flush()
+	if err != nil {
+		panic(err)
+	}
+}
+
+func benchmarkCuckooPanmari(strs [][]byte) {
+	// Default config: https://github.com/panmari/cuckoofilter
+	// b (bucket size): 4
+	// f (fingerprint): 16
+	// False positive rate: 0.0001
+	numKeys := uint(len(strs)) // num of keys the filter will store
+
+	cuckoo := cuckooPan.NewFilter(numKeys)
+
+	for _, str := range strs {
+		cuckoo.Insert(str) // Also try insert unique
+	}
+
+	fname := "cuckoo_filter_pan.bin"
+	file, err := os.OpenFile(fname, os.O_WRONLY|os.O_CREATE, 0644)
+	if err != nil {
+		panic(err)
+	}
+	defer file.Close()
+
+	writer := bufio.NewWriter(file)
+
+	data := cuckoo.Encode()
+
+	_, err = writer.Write(data)
+	if err != nil {
+		panic(err)
+	}
+
+	err = writer.Flush()
+	if err != nil {
+		panic(err)
+	}
+}
+
+// // Function to deserialize binaryfuse8 filter from a file
+// func deserializeBinaryFuse8Filter(filename string) (*xorfilter.BinaryFuse8, error) {
+//     file, err := os.Open(filename)
+//     if err != nil {
+//         return nil, err
```

Comment:
nit: Don't add commented-out code. Delete it, or uncomment it

Reply:
We cannot uncomment it since it's not used and gives lint error so have deleted it. Have copied it in the PR description and referred the PR link for reference.